pkg/asset/cluster/tfvars.go

```diff
@@ -447,6 +447,16 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			workerConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*ibmcloudprovider.IBMCloudMachineProviderSpec)
 		}
 
+		// Get dedicated host info
+		var masterDedicatedHosts []ibmcloud.DedicatedHost
+		if installConfig.Config.ControlPlane.Platform.IBMCloud != nil {
+			masterDedicatedHosts = installConfig.Config.ControlPlane.Platform.IBMCloud.DedicatedHosts
```

Comment:
We want to get it from the install-config. Ok will add validation there.

Reply:
Are you for sure sure? Typically, at this stage, the machines and machinesets are the source of truth for machine-specific details rather than the install config.

Reply:
The caveat here is that once the machine/machinesets are generated, all it has in the API spec is the "name" of the dedicated host. This makes it difficult to determine if the dedicated host needed to be created or not.

Reply:
Ah, fair point. What do you expect the behavior to be if the user manually modifies the machines/machinesets manifests?

Reply:
Hm I haven't accounted for that use case.. In that case, is it safe to assume if the dedicated host name in the machine/machineset is the default name including the cluster_id, it needs to be created? Not sure if there would be a case where user would have access to the cluster_id (with the hash suffix) to manually create a dedicated host with that name.

Reply:
I'm probably overthinking this. I think we are fine with not supporting the user making manual modifications to the dedicated hosts in the machine and machinesets. The supported way to set the dedicated hosts is via the install config. So, the only additional part needed here is to account for the use case where the dedicated hosts are set in the default machine pool and not in either the control plane or compute pools.

Reply:
Got it - thanks for the feedback. Will modify to account for the default machine pool.